### PR TITLE
Fix issue with `PacketReceiptNotFound` error returned by `get_packet_receipt` method 

### LINKED ISSUE #409: Accepting `PacketReceiptNotFound` error returned by `get_packet_receipt` (under `val_exec_ctx` feature)
## Bug Summary

Trying a token transfer with "basecoin-rs" the IBC-rs caught an error. 
It is when the `MsgRecvPacket` message goes through the `validate()` step under the `val_exec_ctx` feature.  
The `PacketReceiptNotFound` error variant resulting from the `get_packet_receipt` method should be accepted

### PR DESCRIPTION
**Pull request by Farhad-Shabani**
Closes: #409 
______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [X] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/ibc/src/core/ics04_channel/handler/recv_packet.rs [hunk 1]
```diff
@@ -128,12 +127,21 @@ pub(crate) mod val_exec_ctx {
...
-
+            ));
+            match packet_rec {
+                Ok(_receipt) => {}
+                Err(ContextError::PacketError(PacketError::PacketReceiptNotFound { sequence }))
+                    if sequence == msg.packet.sequence => {}
+                Err(_) => {
+                    return Err(ContextError::PacketError(
+                        PacketError::ImplementationSpecific,
+                    ))
```
Comment (plafer) [inline crates/ibc/src/core/ics04_channel/handler/recv_packet.rs]: why discard the original error here?

Comment (Farhad-Shabani) [reply crates/ibc/src/core/ics04_channel/handler/recv_packet.rs]: to comply with [here](https://github.com/cosmos/ibc-rs/blob/HEAD/crates/ibc/src/core/ics04_channel/handler/recv_packet.rs#L323). Maybe I should fix this line instead?

Comment (plafer) [reply crates/ibc/src/core/ics04_channel/handler/recv_packet.rs]: yes; I don't like the "implementation specific" error

**Review by plafer — APPROVED** [review]
🙏

File: crates/ibc/src/core/ics04_channel/handler/recv_packet.rs (hunk 1 above)
Comment (Farhad-Shabani) [reply crates/ibc/src/core/ics04_channel/handler/recv_packet.rs]: Agree, done!

**Pull request merged**